tabular/src/autogluon/tabular/models/fastainn/tabular_nn_fastai.py

```diff
@@ -139,10 +141,13 @@ def preprocess_train(self, X_train, y_train, X_val, y_val, **kwargs):
         df_train, train_idx, val_idx = self._generate_datasets(X_train, y_train_norm, X_val, y_val_norm)
         label_class = FloatList if self.problem_type == REGRESSION else None
         procs = [FillMissing, Categorify, Normalize]
+
+        # additional workers are helping only when fork is enabled; in other mp modes, communication overhead reduces performance
+        num_workers = defaults.cpus if is_fork_enabled() else 0
         data = (TabularList.from_df(df_train, path=self.path, cat_names=self.cat_columns, cont_names=self.cont_columns, procs=procs)
                 .split_by_idxs(train_idx, val_idx)
                 .label_from_df(cols=LABEL, label_cls=label_class)
-                .databunch(bs=self.params['bs'] if len(X_train) > self.params['bs'] else 32))
+                .databunch(bs=self.params['bs'] if len(X_train) > self.params['bs'] else 32, num_workers=num_workers))
```

Comment:
Is this a speedup from mainline when we aren't doing HPO?

Reply:
It's a speedup for HPO, for non-HPO the performance will be the same as it was before this PR (unchanged)